src/CMakeLists.txt

```diff
@@ -620,7 +620,7 @@ if(WITH_LIBCEPHFS)
   set(libcephfs_srcs libcephfs.cc)
   add_library(cephfs ${CEPH_SHARED} ${libcephfs_srcs})
   target_link_libraries(cephfs PRIVATE client ceph-common
-    ${CRYPTO_LIBS} ${EXTRALIBS})
+    ${CRYPTO_LIBS} ${EXTRALIBS} heap_profiler)
```

Comment:
It may be caused by this. Since you're linking `heap_profiler` in the `client` library, this one is unnecessary. It may be the linker is including the `heap_profiler` library twice.